src/main/java/org/folio/services/validator/registry/ValidatorRegistryServiceImpl.java

```diff
@@ -0,0 +1,216 @@
+  private static final String RULE_ID_FIELD = "ruleId";
+  private static final String RULE_ID_JSONB_FIELD = "'ruleId'";
+  private static final String STATE_JSONB_FIELD = "'state'";
+  private static final String TYPE_JSONB_FIELD = "'type'";
+
+  private final Vertx vertx;
+
+  public ValidatorRegistryServiceImpl(Vertx vertx) {
+    this.vertx = vertx;
+  }
+
+  /**
+   * Returns all rules for tenant
+   *
+   * @param tenantId           tenant id
+   * @param asyncResultHandler result handler
+   * @return a reference to this, so the API can be used fluently
+   */
+  @Override
+  public ValidatorRegistryService getAllTenantRules(String tenantId, Handler<AsyncResult<JsonObject>> asyncResultHandler) {
+    try {
+      CQL2PgJSON cql2pgJson = new CQL2PgJSON(VALIDATION_RULES_TABLE_NAME + ".jsonb");
+      CQLWrapper cql = new CQLWrapper();
+      cql.setField(cql2pgJson);
+      String[] fieldList = {"*"};
+      PostgresClient.getInstance(vertx, tenantId).get(VALIDATION_RULES_TABLE_NAME, Rule.class, fieldList, cql, true, false, getReply -> {
+        if (getReply.failed()) {
+          logger.error("Error while querying the db to get all tenant rules", getReply.cause());
+          asyncResultHandler.handle(Future.failedFuture(getReply.cause()));
+        } else {
+          RuleCollection rules = new RuleCollection();
+          List<Rule> ruleList = (List<Rule>) getReply.result().getResults();
+          rules.setRules(ruleList);
+          rules.setTotalRecords(ruleList.size());
+          asyncResultHandler.handle(Future.succeededFuture(JsonObject.mapFrom(rules)));
+        }
+      });
+    } catch (Exception e) {
+      logger.error("Error while getting all tenant rules", e.getCause());
+      asyncResultHandler.handle(Future.failedFuture(e.getCause()));
+    }
+    return this;
+  }
+
+  /**
+   * Creates rule for tenant with specified id
+   *
+   * @param tenantId           tenant id
+   * @param validationRule     rule to save
+   * @param asyncResultHandler result handler
+   * @return a reference to this, so the API can be used fluently
+   */
+  @Override
+  public ValidatorRegistryService createTenantRule(String tenantId, JsonObject validationRule, Handler<AsyncResult<JsonObject>> asyncResultHandler) {
+    try {
+      String id = UUID.randomUUID().toString();
+      validationRule.put(RULE_ID_FIELD, id);
+      PostgresClient.getInstance(vertx, tenantId).save(VALIDATION_RULES_TABLE_NAME, id, validationRule.mapTo(Rule.class), postReply -> {
+        if (postReply.failed()) {
+          logger.error("Error while saving the rule to the db", postReply.cause());
+          asyncResultHandler.handle(Future.failedFuture(postReply.cause()));
+        } else {
+          asyncResultHandler.handle(Future.succeededFuture(validationRule));
+        }
+      });
+    } catch (Exception e) {
+      logger.error("Error while creating new tenant rule", e.getCause());
```

Comment:
e.getCause() may be null if thrown from PostgresClient.getInstance or PostgresClient.getInstance(vertx, tenantId).save